File: frontend/src/types/job-posting.ts
export type PostingStatus = "Published" | "InProgress" | "Closed" | "Cancelled";
export type SettlementStatus = "Unsettled" | "Settled";

export interface JobPosting {
  id: number;
  customer_id: number;
  salary: number;
  description: string;
  employer_fee_rate: number | null;
  settlement_status: SettlementStatus;
  settlement_amount: number | null;
  settlement_memo: string | null;
  posting_status: PostingStatus;
  is_favorite: boolean;
  created_at: string;
  updated_at: string;
  deleted_at: string | null;
}

export interface JobSeekingPosting {
  id: number;
  customer_id: number;
  desired_salary: number;
  description: string;
  preferred_location: string;
  employee_fee_rate: number | null;
  settlement_status: SettlementStatus;
  settlement_amount: number | null;
  settlement_memo: string | null;
  posting_status: PostingStatus;
  is_favorite: boolean;
  created_at: string;
  updated_at: string;
  deleted_at: string | null;
}

export interface JobPostingMemo {
  id: number;
  job_posting_id: number;
  memo_content: string;
  created_by: number | null;
  created_at: string;
  updated_at: string;
  deleted_at: string | null;
}

export interface JobSeekingMemo {
  id: number;
  job_seeking_posting_id: number;
  memo_content: string;
  created_by: number | null;
  created_at: string;
  updated_at: string;
  deleted_at: string | null;
}

export interface CreateJobPostingRequest {
  customer_id: number;
  salary: number;
  description: string;
  employer_fee_rate?: number | null;
}

export interface UpdateJobPostingRequest {
  salary?: number;
  description?: string;
  employer_fee_rate?: number | null;
  settlement_status?: SettlementStatus;
  settlement_amount?: number | null;
  settlement_memo?: string | null;
  posting_status?: PostingStatus;
  is_favorite?: boolean;
}

export interface CreateJobSeekingRequest {
  customer_id: number;
  desired_salary: number;
  description: string;
  preferred_location: string;
  employee_fee_rate?: number | null;
}

export interface UpdateJobSeekingRequest {
  desired_salary?: number;
  description?: string;
  preferred_location?: string;
  employee_fee_rate?: number | null;
  settlement_status?: SettlementStatus;
  settlement_amount?: number | null;
  settlement_memo?: string | null;
  posting_status?: PostingStatus;
  is_favorite?: boolean;
}

File: backend/src/models/job_posting.rs
use chrono::NaiveDateTime;
use rust_decimal::Decimal;
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, sqlx::Type, PartialEq)]
#[sqlx(type_name = "text", rename_all = "lowercase")]
#[serde(rename_all = "PascalCase")]
pub enum PostingStatus {
    #[sqlx(rename = "published")]
    Published,
    #[sqlx(rename = "in_progress")]
    #[serde(rename = "InProgress")]
    InProgress,
    #[sqlx(rename = "closed")]
    Closed,
    #[sqlx(rename = "cancelled")]
    Cancelled,
}

#[derive(Debug, Clone, Serialize, Deserialize, sqlx::Type, PartialEq)]
#[sqlx(type_name = "text", rename_all = "lowercase")]
#[serde(rename_all = "PascalCase")]
pub enum SettlementStatus {
    #[sqlx(rename = "unsettled")]
    Unsettled,
    #[sqlx(rename = "settled")]
    Settled,
}

#[derive(Debug, Clone, Serialize, Deserialize, sqlx::FromRow)]
pub struct JobPosting {
    pub id: i64,
    pub customer_id: i64,
    pub salary: Decimal,
    pub description: String,
    pub employer_fee_rate: Option<Decimal>,
    pub settlement_status: SettlementStatus,
    pub settlement_amount: Option<Decimal>,
    pub settlement_memo: Option<String>,
    pub posting_status: PostingStatus,
    pub is_favorite: bool,
    pub created_at: NaiveDateTime,
    pub updated_at: NaiveDateTime,
    pub deleted_at: Option<NaiveDateTime>,
}

#[derive(Debug, Clone, Deserialize)]
pub struct CreateJobPostingRequest {
    pub customer_id: i64,
    pub salary: Decimal,
    pub description: String,
    pub employer_fee_rate: Option<Decimal>,
}

#[derive(Debug, Clone, Deserialize)]
pub struct UpdateJobPostingRequest {
    pub salary: Option<Decimal>,
    pub description: Option<String>,
    pub employer_fee_rate: Option<Decimal>,
    pub settlement_status: Option<SettlementStatus>,
    pub settlement_amount: Option<Decimal>,
    pub settlement_memo: Option<String>,
    pub posting_status: Option<PostingStatus>,
    pub is_favorite: Option<bool>,
}

#[derive(Debug, Clone, Serialize, Deserialize, sqlx::FromRow)]
pub struct JobSeekingPosting {
    pub id: i64,
    pub customer_id: i64,
    pub desired_salary: Decimal,
    pub description: String,
    pub preferred_location: String,
    pub employee_fee_rate: Option<Decimal>,
    pub settlement_status: SettlementStatus,
    pub settlement_amount: Option<Decimal>,
    pub settlement_memo: Option<String>,
    pub posting_status: PostingStatus,
    pub is_favorite: bool,
    pub created_at: NaiveDateTime,
    pub updated_at: NaiveDateTime,
    pub deleted_at: Option<NaiveDateTime>,
}

#[derive(Debug, Clone, Deserialize)]
pub struct CreateJobSeekingPostingRequest {
    pub customer_id: i64,
    pub desired_salary: Decimal,
    pub description: String,
    pub preferred_location: String,
    pub employee_fee_rate: Option<Decimal>,
}

#[derive(Debug, Clone, Deserialize)]
pub struct UpdateJobSeekingPostingRequest {
    pub desired_salary: Option<Decimal>,
    pub description: Option<String>,
    pub preferred_location: Option<String>,
    pub employee_fee_rate: Option<Decimal>,
    pub settlement_status: Option<SettlementStatus>,
    pub settlement_amount: Option<Decimal>,
    pub settlement_memo: Option<String>,
    pub posting_status: Option<PostingStatus>,
    pub is_favorite: Option<bool>,
}
